staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -164,6 +170,25 @@ func (p *Patcher) patchSimple(obj runtime.Object, modified []byte, namespace, na
 	}
 
 	patchedObj, err := p.Helper.Patch(namespace, name, patchType, patch, nil)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	patchedMeta, err := meta.Accessor(patchedObj)
+
+	if err != nil {
+		// Ignore error? request did go through...
+		return patch, patchedObj, nil
+	}
+
+	// The server side may detect a no-op patch and do nothing with it.
+	// 	If the returned object has the same resource version then the server did
+	// 	not apply any new changes and effectively nothing has happened.
+	if patchedMeta.GetResourceVersion() == objMeta.GetResourceVersion() {
```

Comment:
Does this `resourceVersion` setting logic https://github.com/kubernetes/kubernetes/blob/66c2f6069ebbfa41d0bef0f3959abda59b8483f1/staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go#L159-L164 have any effect on this check?.

As far as I can see only diff uses it and it already uses current object's resource version and any other code does not use it. I think, it might not be an issue because currently we are always saying `configured` and worst case we continue to do so.

Reply:
Yeah, so there is a possibility for a race-condition here. If someone else changes the object in the mean-time, it'll show as configured when it's not.

/shrug

Not sure what's best? I suspect that showing something that feels "random" isn't great.

Reply:
I considered it an improvement since there are some patches that are affected by this issue (like those involving deployments due to `retainKeys`), and other that are not (those that are fully specified). There is an argument to be made that it "seems random" which schemas are affected.

However I am inclined to agree that behavior that may change over time is worse for UX than behavior that changes depending on the input. 

But it still feels like kubectl is lying to me when it says a object is "configured" when in fact it is "maybe configured".

Reply:
> Does this resourceVersion setting logic have any effect on this check?

It should not since the new RV is always changed if there was a change to the underlying object.